resample/bootstrap.py

```diff
@@ -0,0 +1,285 @@
+"""
+Bootstrap resampling.
+"""
+from typing import Callable, Optional, Tuple, Sequence, Union
+import numpy as np
+from scipy import stats
+from resample.jackknife import jackknife as _jackknife
+from resample.empirical import quantile_gen as _quantile_gen
```

Comment:
I'd personally rather use something like `quantile_function` or `quantile_fn` since the `_gen` sounds a bit peculiar (it makes it sound like it's a generator but that's not what we mean). Similar to `confidence_interval` a user can always abbreviate `quantile_function` to something shorter.

Reply:
I chose `..._gen` because Scipy does uses it. They have `poisson_gen` to generate a `poisson` object, for instance (in stats/_discrete_distns.py). But if you prefer `_fn` then I will change this.

Reply:
Changed to _gen.